src/compartment.c

```diff
@@ -75,17 +75,20 @@ comp_from_elf(char *filename, char **entry_points, size_t entry_point_count,
     assert(new_comp->fd != -1);
     do_pread(new_comp->fd, &comp_ehdr, sizeof(Elf64_Ehdr), 0);
     new_comp->elf_type = comp_ehdr.e_type;
-    assert(new_comp->elf_type == ET_DYN || new_comp->elf_type == ET_EXEC);
+    if (new_comp->elf_type != ET_DYN)
+    {
+        errx(1, "Only supporting DYN ELF types!");
```

Comment:
I guess this means "Only DYN ELF is supported"?

Reply:
Yes, as per the `Type` property in `readelf -h`. Should I reword?

Reply:
Done in https://github.com/capablevms/CHERI-ELF-comp/pull/19/commits/18f6c8f12fae70a37d72d96e4776809cef17fad8.